lib/nexpose/scheduled_backup.rb

```diff
@@ -0,0 +1,86 @@
+module Nexpose
+# Constants useful across the Nexpose module.
+# Configuration structure for scheduled backups.
+  class ScheduledBackup < APIObject
+    require 'json'
+    include JsonSerializer
+
+    # Whether or not this schedule is enabled. Defaults to true if not set
+    attr_accessor :enabled
+    # Valid schedule types: daily, hourly, monthly-date, monthly-day, weekly.
+    attr_accessor :schedule_type
+    # The repeat interval based upon type.
+    attr_accessor :schedule_interval
+    # The earliest date to generate the report on (in ISO 8601 format).
+    attr_accessor :schedule_start
+    # The description of the backup. Defaults to nil if not set
+    attr_accessor :description
+    # Whether the backup will be platform independent or not. Defaults to true if not set
+    attr_accessor :platform_independent
+    # Whether the backup should pause all local scans or wait for local scans to complete. Defaults to true if not set
+    attr_accessor :pause_local_scans
+    # Cancellation window in minutes for scheduled DB maintenance. Defaults to 0 if not set
+    attr_accessor :cancellation_period
```

Comment:
Yeh like Gavin said, the cancellation_period is basically a set amount of time we will wait for running/pausing scans that block a backup/maintenance from happening. After that time the backup/maintenance doesn't happen. An example use case could be a customer has a specific window in which they can run the backup but may wish to abandon the backup if it hasn't started after x amount of minutes. Kinda different from non-scheduled backups which will run after all scans have completed, whether that's 1 hour or 24 hours etc.